### PR TITLE
feat:XA transaction basic shelf construction

### PR DESCRIPTION
**Pull request by chuntaojun**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/transaction/trx_log.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	txr_id varchar(255) NOT NULL,
+    tenant varchar(255) NOT NULL,
+	server_id int(10) UNSIGNED NOT NULL,
+	status int(10) NOT NULL,
+	participant varchar(500),
+	start_time timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP
+	update_time timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+	PRIMARY KEY (log_id),
+	UNIQUE KEY (txr_id)
+  ) ENGINE = InnoDB CHARSET = utf8
```
Comment (dongzl) [inline pkg/runtime/transaction/trx_log.go:58]: Suggest format the  sql content, and add comment for sql column

File: test/integration_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func TestSuite(t *testing.T) {
 		WithMySQLDatabase("employees"),
 		WithConfig("../integration_test/config/db_tbl/config.yaml"),
 		WithScriptPath("../scripts"),
-		// WithDevMode(), // NOTICE: UNCOMMENT IF YOU WANT TO DEBUG LOCAL ARANA SERVER!!!
+		//WithDevMode(), // NOTICE: UNCOMMENT IF YOU WANT TO DEBUG LOCAL ARANA SERVER!!!
```
Comment (dongzl) [inline test/integration_test.go:57]: maybe need to keep the space

**Review by dongzl — CHANGES_REQUESTED** [review]

File: integration_test/config/db/config.yaml [hunk 3]
```diff
@@ -27,6 +27,13 @@ data:
           password: "123456"
         - username: arana
           password: "123456"
+      sysDB:
```
Comment (dongzl) [inline integration_test/config/db/config.yaml:30]: we can discuss a suitable name.

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/transaction/trx_log.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		stateVal,
+		participantsVal,
+	}
+	_, _, err = gm.sysDB.Call(context.Background(), insSql, args...)
+	return err
+}
+
+// DeleteTxLog Delete transaction log
+func (gm *TxLogManager) DeleteTxLog(l TrxLog) error {
+	delSql := `
```
Comment (dongzl) [inline pkg/runtime/transaction/trx_log.go:108]: Maybe defined as a string?

**Review by dongzl — CHANGES_REQUESTED** [review]

**Comment by maronghe** [comment]
需要rebase一下吗

File: integration_test/config/db/config.yaml (hunk 3 above)
Comment (jjeffcaii) [inline integration_test/config/db/config.yaml:30]: How about underscore style `sys_db`?

File: pkg/runtime/transaction/deadlock.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	timer       *time.Timer
+	hasDeadLock int32
+}
+
+// Start run deadlock detection dog, can set how long the delay starts to execute
+func (xd *xaDeadLockDog) Start(ctx context.Context, delay time.Duration, tx runtime.CompositeTx) {
+	_ctx, cancel := context.WithCancel(ctx)
+
+	xd.ctx = _ctx
+	xd.cancel = cancel
```
Comment (jjeffcaii) [inline pkg/runtime/transaction/deadlock.go:61]: Use `xd.ctx,xd.cancel = context.WithCancel(ctx)`  instead.

File: pkg/runtime/transaction/hook.go [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+		runtime.TrxCommitting: xh.onCommitting,
+		runtime.TrxCommitted:  xh.onCommitted,
+		runtime.TrxAborting:   xh.onAborting,
+		runtime.TrxRollback:   xh.onRollbackOnly,
+		runtime.TrxRolledBack: xh.onRolledBack,
+	}
+
+	xh.trxMgr = trxMgr
+	xh.trxLog = &TrxLog{
+		Participants: []TrxParticipant{},
```
Comment (jjeffcaii) [inline pkg/runtime/transaction/hook.go:56]: No need to initialize a slice.

File: pkg/runtime/transaction/types.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	RemoteAddr string
+	Schema     string
+}
+
+type dBOperation string
+
+const (
+	Like           dBOperation = "LIKE"
+	Equal          dBOperation = "="
+	NotEqual       dBOperation = "!="
```
Comment (jjeffcaii) [inline pkg/runtime/transaction/types.go:45]: I think it will be better to use `<>` instead.

**Review by jjeffcaii — COMMENTED** [review]
pls fix

**Review by jjeffcaii — APPROVED** [review]

**Review by Mulavar — APPROVED** [review]

**Review by dongzl — APPROVED** [review]

**Pull request merged**